.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "master"
+
+jobs:
+  format_and_lint:
+    name: Format and lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install latest nightly
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            override: true
+            components: rustfmt, clippy
+      - uses: Swatinem/rust-cache@v1
```

Comment:
As pointed out below, consider using `actions/cache@v2` instead of `Swatinem/rust-cache@v1`